exec/java-exec/src/main/java/org/apache/drill/exec/physical/EndpointAffinity.java

```diff
@@ -96,6 +96,42 @@ public boolean isAssignmentRequired() {
   }
 
   @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    long temp;
+    temp = Double.doubleToLongBits(affinity);
+    result = prime * result + (int) (temp ^ (temp >>> 32));
+    result = prime * result + ((endpoint == null) ? 0 : endpoint.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (!(obj instanceof EndpointAffinity)) {
+      return false;
+    }
+    EndpointAffinity other = (EndpointAffinity) obj;
+    if (Double.doubleToLongBits(affinity) != Double.doubleToLongBits(other.affinity)) {
+      return false;
+    }
+    if (endpoint == null) {
+      if (other.endpoint != null) {
+        return false;
+      }
+    } else if (!endpoint.equals(other.endpoint)) {
```

Comment:
DrillbitEndpoint is a protobuf Message, so the equals/hashCode methods are provided by protobuf API (by AbstractMessage if I'm correct)